pkg/validation/validate_cluster.go

```diff
@@ -54,6 +60,60 @@ type ValidationNode struct {
 	Status   v1.ConditionStatus `json:"status,omitempty"`
 }
 
+type ValidateClusterRetries struct {
+	Cluster         *api.Cluster
+	Clientset       simple.Clientset
+	Interval        time.Duration
+	ValidateRetries int
+	K8sClient       kubernetes.Interface
+}
+
+func (v ValidateClusterRetries) GetInstanceGroupsAndValidateCluster() error {
+	// get the new list of ig
+	list, err := v.Clientset.InstanceGroupsFor(v.Cluster).List(metav1.ListOptions{})
+	if err != nil {
+		return fmt.Errorf("unable to get instance groups: %v", err)
+	}
+
+	// validate the cluster
+	if err := v.ValidateClusterWithRetries(list); err != nil {
+		return fmt.Errorf("unable to validate cluster: %v", err)
+	}
+	return nil
+}
+
+// ValidateClusterWithRetries runs our validation methods on the K8s Cluster x times and then fails.
+func (v ValidateClusterRetries) ValidateClusterWithRetries(instanceGroupList *api.InstanceGroupList) (err error) {
+
+	// TODO - We are going to need to improve Validate to allow for more than one node, not master
+	// TODO - going down at a time.
+	for i := 0; i <= v.ValidateRetries; i++ {
+
+		if _, err = ValidateCluster(v.Cluster.ObjectMeta.Name, instanceGroupList, v.K8sClient); err != nil {
+			glog.Infof("Cluster did not validate, and waiting longer: %v.", err)
+			time.Sleep(v.Interval / 2)
+		} else {
+			glog.Infof("Cluster validated.")
+			return nil
+		}
+
+	}
+
+	// FIXME handle force and cloud only
+	// for loop is done, and did not end when the cluster validated
+	return fmt.Errorf("cluster validation failed: %v", err)
+}
+
+func (v *ValidateClusterRetries) ValidateCluster(instanceGroupList *api.InstanceGroupList) error {
+
+	if _, err := ValidateCluster(v.Cluster.ObjectMeta.Name, instanceGroupList, v.K8sClient); err != nil {
+		return fmt.Errorf("cluster %q did not pass validation: %v", v.Cluster.ObjectMeta.Name, err)
+	}
+
+	return nil
+
+}
+
 // ValidateCluster validate a k8s cluster with a provided instance group list
 func ValidateCluster(clusterName string, instanceGroupList *kops.InstanceGroupList, clusterKubernetesClient kubernetes.Interface) (*ValidationCluster, error) {
```

Comment:
Seems like it would be easy to add tests for this since we already have a fake kubenetes client